lib/src/lib.rs

```diff
@@ -1,19 +1,49 @@
 #![warn(
     clippy::all,
-    clippy::doc_markdown,
+    clippy::await_holding_lock,
     clippy::dbg_macro,
-    clippy::todo,
+    clippy::debug_assert_with_mut_call,
+    clippy::doc_markdown,
     clippy::empty_enum,
     clippy::enum_glob_use,
-    clippy::pub_enum_variant_names,
-    clippy::mem_forget,
-    clippy::use_self,
+    clippy::exit,
+    clippy::explicit_into_iter_loop,
     clippy::filter_map_next,
-    clippy::needless_continue,
+    clippy::fn_params_excessive_bools,
+    clippy::if_let_mutex,
+    clippy::imprecise_flops,
+    clippy::inefficient_to_string,
+    clippy::large_types_passed_by_value,
+    clippy::let_unit_value,
+    clippy::linkedlist,
+    clippy::lossy_float_literal,
+    clippy::macro_use_imports,
+    clippy::map_err_ignore,
+    clippy::map_flatten,
+    clippy::map_unwrap_or,
+    clippy::match_on_vec_items,
+    clippy::match_same_arms,
+    clippy::match_wildcard_for_single_variants,
+    clippy::mem_forget,
+    clippy::mismatched_target_os,
     clippy::needless_borrow,
-    rust_2018_idioms,
+    clippy::needless_continue,
+    clippy::option_option,
+    clippy::pub_enum_variant_names,
+    clippy::ref_option_ref,
+    clippy::rest_pat_in_fully_bound_structs,
+    clippy::string_add_assign,
+    clippy::string_add,
+    clippy::string_to_string,
+    clippy::suboptimal_flops,
+    clippy::todo,
+    clippy::unimplemented,
+    clippy::unnested_or_patterns,
+    clippy::unused_self,
+    clippy::verbose_file_reads,
     future_incompatible,
-    nonstandard_style
+    nonstandard_style,
+    rust_2018_idioms
```

Comment:
Make sure to copy the entire file contents of https://github.com/EmbarkStudios/rust-ecosystem/blob/main/lints.rs for the lints so one gets the `/// BEGIN` / `/// END` lines as well so one sees the version and source of it. and is easier to upgrade later

Reply:
Will do